src/test/java/bio/terra/service/snapshot/SnapshotIntegrationTest.java

```diff
@@ -276,4 +281,50 @@ private SnapshotRequestModel snapshotByQueryRequestModel(DatasetModel dataset) t
                 + ".sample.id ='sample6'");
     return requestModel;
   }
+
+  @Test
+  public void testCreateSnapshotWithPolicies() throws Exception {
+    DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+    String datasetName = dataset.getName();
+    SnapshotRequestModel requestModel =
+        jsonLoader.loadObject("ingest-test-snapshot-fullviews.json", SnapshotRequestModel.class);
+    // swap in the correct dataset name (with the id at the end)
+    requestModel.getContents().get(0).setDatasetName(datasetName);
+
+    List<String> stewards = List.of(steward().getEmail(), admin().getEmail());
+    String readerEmail = reader().getEmail();
+    List<String> readersWithDuplicates = List.of(readerEmail, readerEmail);
+    String discovererEmail = discoverer().getEmail();
+    SnapshotRequestModelPolicies policiesRequest =
+        new SnapshotRequestModelPolicies()
+            .stewards(stewards)
+            .readers(readersWithDuplicates)
+            .addDiscoverersItem(discovererEmail);
+    requestModel.setPolicies(policiesRequest);
+
+    SnapshotSummaryModel snapshotSummary =
+        dataRepoFixtures.createSnapshotWithRequest(steward(), datasetName, profileId, requestModel);
+    TimeUnit.SECONDS.sleep(10);
+    UUID snapshotId = snapshotSummary.getId();
+    createdSnapshotIds.add(snapshotId);
+
+    Map<String, List<String>> rolesToPolicies =
+        dataRepoFixtures.retrieveSnapshotPolicies(steward(), snapshotId).getPolicies().stream()
+            .collect(Collectors.toMap(PolicyModel::getName, PolicyModel::getMembers));
+
+    assertThat(
+        "All specified stewards added on snapshot creation",
+        rolesToPolicies.get(IamRole.STEWARD.toString()),
+        containsInAnyOrder(stewards.toArray()));
+
+    assertThat(
+        "Reader added on snapshot creation, duplicates removed without error",
```

Comment:
@nmalfroy -- in this integration test, we pass duplicate readers in the request, and verify that when we fetch the policies, the readers have been deduplicated.